llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1523,9 +1523,12 @@ unsigned RISCVInstrInfo::getInstSizeInBytes(const MachineInstr &MI) const {
   case TargetOpcode::PATCHPOINT:
     // The size of the patchpoint intrinsic is the number of bytes requested
     return PatchPointOpers(&MI).getNumPatchBytes();
-  case TargetOpcode::STATEPOINT:
+  case TargetOpcode::STATEPOINT: {
     // The size of the statepoint intrinsic is the number of bytes requested
-    return StatepointOpers(&MI).getNumPatchBytes();
+    unsigned NumBytes = StatepointOpers(&MI).getNumPatchBytes();
+    // A statepoint is at least a PseudoCALL
```

Comment:
Is this right for all target types?

Reply:
This comment is slightly off, so I will replace it with `// No patch bytes means a PseudoCall is emitted`. Otherwise, using `getNumPatchBytes` is always correct as `NOPS` are emitted to match the requested patch size.